networkx/generators/classic.py

```diff
@@ -706,6 +707,59 @@ def star_graph(n, create_using=None):
     return G
 
 
+@nodes_or_number([0, 1])
+@nx._dispatch(graphs=None)
+def tadpole_graph(m, n, create_using=None):
+    """Returns the Tadpole Graph; `C_m` connected to `P_n`.
+
+    Parameters
+    ----------
+    m, n : int or iterable container of nodes (default = 0)
+        If an integer, nodes are from `range(m)` and `range(m,m+n)`.
+        If a container of nodes, those nodes appear in the graph.
```

Comment:
Why not allow for `m=0` or `m=1` ? 
I changed the code so `m=0` and `m=1` are allowed, too.

Reply:
Things may get ambiguous...
For example, for `m=1`, I think a cycle graph of 1 node just consists of a node without any links (but you suggest a self-loop). To avoid these ambiguities, perhaps we should enforce `m >=2` ?

Reply:
Actually,  `nx.cycle_graph(1)` returns a graph with one edge `(0, 0)`. 
I'm also fine with enforcing `m>=2` if you prefer that. I kind of like the idea that a very small head on the tadpole leads to a self-loop. But I don't feel strongly about it.